rust-mcd/src/error.rs

```diff
@@ -31,14 +31,28 @@ pub struct Error {
     inner: mcd_error_info_st,
 }
 
+impl Error {
+    pub fn error_code(&self) -> McdErrorCode {
+        McdErrorCode::from_code(self.inner.error_code)
+    }
+
+    pub fn event_error_code(&self) -> EventError {
+        EventError::from_bits_retain(self.inner.error_events)
+    }
+}
+
 impl std::error::Error for Error {}
 
 impl Display for Error {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
         let info = unsafe { CStr::from_ptr(&self.inner.error_str[0] as *const i8) }
```

Comment:
This could be changed to avoid the invariants `CStr::from_ptr()` requires:
```rust
// SAFETY:
// Arrays of u8 share the same layout as arrays of i8 and i8 can be transmuted to u8 safely.
let error_str: &[u8; 256] = unsafe { std::mem::transmute(&self.inner.error_str) };
let info = CStr::from_bytes_until_nul(error_str)
    .unwrap()
    .to_str()
    .unwrap();
```

Depending on performance needs, this could also be achieved without unsafe via `map()`.

Reply:
Good point!